app/models/binary/DataSet.scala

```diff
@@ -163,11 +163,18 @@ class DataSetDAO @Inject()(sqlClient: SQLClient,
       parsed <- parseFirst(r, id)
     } yield parsed
 
-  def findAllWithSearch(folderIdOpt: Option[ObjectId], searchQuery: Option[String])(
+  def findAllWithSearch(folderIdOpt: Option[ObjectId], searchQuery: Option[String], recursive: Boolean = false)(
       implicit ctx: DBAccessContext): Fox[List[DataSet]] =
     for {
       accessQuery <- readAccessQuery
-      folderPredicate = folderIdOpt.map(folderId => s"_folder = '$folderId'").getOrElse("true")
+      folderPredicate = folderIdOpt
+        .map(
+          folderId =>
+            if (recursive)
```

Comment:
Since there are three possible code paths here now, I’d replace the map/if/else/getOrElse with a match expression

```suggestion
      folderPredicate = folderIdOpt match {
        case Some(folderId) if (recursive) =>
```

and then `case Some(folderId) => … `  and then `case None => "true"`